contrib/dracut/90zfs/mount-zfs.sh.in

```diff
@@ -56,6 +56,27 @@ ZFS_DATASET="${ZFS_DATASET:-${root#zfs:}}"
 ZFS_POOL="${ZFS_DATASET%%/*}"
 
 if import_pool "${ZFS_POOL}" ; then
+	# Load keys if we can or if we need to
+	if [ $(zpool list -H -o feature@encryption) == 'active' ]; then
```

Comment:
You should specify the pool name here so that you get just one pool. You might want error handling when the pool is missing.

Reply:
I might be able to derive that from `ZFS_DATASET`